backend/internal/usecase/transaction.go

```diff
@@ -228,19 +229,20 @@ func (uc *TransactionUsecase) QueryTransactionDetailsByTxHash(ctx context.Contex
 			}
 		} else if tse.TxStatus.TxStatusVal == data.TxStatusValUnshieldedinclusion {
 			txStatus = UnshieldedInclusion
-			effectiveInclusionTime = tse.DecryptedTxCreatedAtUnix
+			effectiveInclusionTime = tse.DecryptedTxUpdatedAtUnix
 			inclusionDelay = effectiveInclusionTime - tse.CreatedAt.Time.Unix()
-			if tse.Slot.Valid {
-				inclusionSlot = tse.Slot.Int64
-			}
 		} else if tse.TxStatus.TxStatusVal == data.TxStatusValInvalid {
 			txStatus = Invalid
+			tse.BlockNumber = pgtype.Int8{Int64: 0, Valid: false}
```

Comment:
I think it would be preferable to set the values to null when these are invalid values, such as block number and slot number for all transactions that do not have the status "Shielded Inclusion", as otherwise, this value can be confusing when read from the API response. Even if we set the valid flag here as false, the API response would still show the block number as 0.